tests/performance/test_performance_testing.py

```diff
@@ -97,44 +102,112 @@ def build_test_suite(
+            asyncio.set_event_loop(self.custom_loop)
+            if self.custom_loop:
+                self.custom_loop.run_forever()
+            raise RuntimeError("No custom_loop set!")
+
+        def __call__(self):
+            evloop = None
+            awaitable = self.coro(*self.args, **self.kwargs)
+            try:
+                evloop = asyncio.get_running_loop()
+            except RuntimeError:
+                pass
+
+            if evloop is None:
+                return asyncio.run(awaitable)
+
+            if not self.custom_loop or not self.thread or not self.thread.is_alive():
+                self.custom_loop = asyncio.new_event_loop()
+                self.thread = threading.Thread(
+                    target=self.start_background_loop, daemon=True
+                )
+                self.thread.start()
+
+            return asyncio.run_coroutine_threadsafe(
+                awaitable, self.custom_loop
+            ).result()
+
+    def _wrapper(func, *args, **kwargs):
+        if asyncio.iscoroutinefunction(func):
+            prepared_function = Sync2Async(func, *args, **kwargs)
+            benchmark.pedantic(prepared_function, rounds=ROUNDS_NUMBER)
+        else:
+            benchmark.pedantic(func, rounds=ROUNDS_NUMBER, args=args, kwargs=kwargs)
+
+    return _wrapper
+
+
+@asynccontextmanager
+async def prepare_tests(contract_class: ContractClass, test_suite: TestSuite):
     with multiprocessing.Manager() as manager:
-        tests_state = SharedTestsState(
-            test_collector_result=TestCollector.Result(test_suites=[test_suite]),
-            manager=manager,
-        )
-        runner = TestRunner(
-            shared_tests_state=tests_state,
-            include_paths=[],
-        )
-
-        # pylint: disable=protected-access
-        await runner._run_test_suite(
-            test_contract=contract,
-            test_suite=test_suite,
+        runner, shared_state, execution_state = await prepare_suite(
+            manager, test_suite, contract_class
         )
+        if not execution_state:
+            return
 
-        tests_left = len(test_suite.test_case_names)
-        while tests_left:
-            test_result = tests_state.get_result()
-            if isinstance(test_result, PassedTestCase):
-                out = sys.stdout
-            else:
-                out = sys.stderr
+        async def run():
+            # pylint: disable=protected-access
+            await runner._invoke_cases(test_suite, execution_state)
 
-            print(test_result.format(), file=out)
-            tests_left -= 1
+        yield run
 
-        assert (
-            not tests_state.any_failed_or_broken()
-        ), "Tests failed! See logs above for more info"
+        wait_for_completion(test_suite, shared_state)
 
 
-def test_deploy_perf(benchmark, tmp_path, basic_contract_path):
+@pytest.mark.asyncio
+async def test_deploy_perf(aio_benchmark, tmp_path, basic_contract_path):
```

Comment:
Marking tests with `@pytest.mark.asyncio` is unnecessary thanks to the following config:

pytest.ini
```
[pytest]
asyncio_mode=auto
```

Reply:
oh thx!